### DISCUSSION TITLE
Access owner and branch of a PR head within GitHub Actions

### DISCUSSION THREAD

**Discussion by pmeier** [community/community#25908] [Actions]
<p>Imagine I have the repository <code>me/myrepo</code>. A PR tries to merge <code>someone-else/myrepo/feature</code> into <code>me/myrepo/master</code>. I need to somehow access <code>someone-else</code> (owner) and <code>feature</code> (branch) within a GitHub Actions workflow.</p>
<p>From the <a href="https://docs.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables" rel="nofollow noopener">environment variables</a> I successfully utilize <code>GITHUB_HEAD_REF</code> for the branch. For the owner I use <code>GITHUB_ACTOR</code> which only works as long as only the person who created the PR adds commits.</p>
<p>Is there a way to get the owner of a PR without relying on the actor that triggered the workflow?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/pmeier">@pmeier</a>,</p>
<p>You can use the expression (<strong>github.event.pull_request.head.repo.owner.login</strong>) to get the owner of the <strong>head</strong> repository from the <a href="https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener"><strong>github</strong> context</a>.</p>
<pre><code class="lang-yaml">on: pull_request
jobs:
  job1:
    . . .
    steps:
      - name: get the owner of the head repository
        run: echo "head owner = ${{ github.event.pull_request.head.repo.owner.login }}"
</code></pre>
<p>Of course, if you feel that this expression is too long, you can map it as a environment variable with a shorter name.</p>
<pre><code class="lang-yaml">on: pull_request

env:
  HEAD_OWNER: ${{ github.event.pull_request.head.repo.owner.login }}

jobs:
  job1:
    . . .
    steps:
      - name: get the owner of the head repository
        run: echo "head owner = ${{ env.HEAD_OWNER }}"
</code></pre>
<p>About the <a href="https://docs.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables" rel="nofollow noopener">default environment variables</a> automatically set by GitHub, most of them (start with the prefix ‘<strong>GITHUB_</strong>’) have the corresponding properties in the github context.<br>
However, not every property in the github context will be mapped as a default environment variable. So, many times you need to use the appropriate expression to fetch the value of the property from the <strong>github</strong> context.<br>
Other <a href="https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#contexts" rel="nofollow noopener"><strong>Contexts</strong></a> are similar.</p>
<p>You can print the <strong>github</strong> context to view more information about the current workflow run and the event that triggered the run.</p>
<pre><code class="lang-yaml">- name: Print the GitHub context
  env:
    GITHUB_CONTEXT: ${{ toJson(github) }}
  run: echo "$GITHUB_CONTEXT"
</code></pre>

**Comment by pmeier** [comment]
<p>Thanks for the tip. I didn’t know about the <code>github</code> context. I’ll give it a try an get back to you if that works for me. Problem might be that I need this to work regardless if I’m within a PR or the PR is merged into <code>master</code>.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/pmeier">@pmeier</a>,</p>
<p>When the workflow is not triggered by the events that related to the PR, if you want to get the information of the PR in this workflow, you can execute the “<a href="https://docs.github.com/en/rest/reference/pulls#get-a-pull-request" rel="nofollow noopener"><strong>Get a pull request</strong></a>” API in the workflow, and extract the information you need from the response of the API.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/pmeier">@pmeier</a>,</p>
<p>How are things going?<br>
Have you tried as my above suggestion that using the expression (<strong>github.event.pull_request.head.repo.owner.login</strong>) to get the owner of the <strong>head</strong> repository from the <strong>github context</strong>, or the “<strong>Get a pull request</strong>” API? Is it helpful for you?</p>
<p>Please try it. Any update, feel free to tell me. Thanks.</p>

**Comment by pmeier** [comment]
<p>Hey <a class="mention" href="/brightran">@brightran</a>, thanks for checking in. Something else came up and I have to handle this first. From what I understand from skimming the links you posted, this is exactly what I need. I’ll just accept your answer as solution for now and ping you if it somehow does not work as I intended it.</p>
<p>Thanks again for your support :slight_smile:</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/pmeier">@pmeier</a>,</p>
<p>You’re welcome.<br>
Any questions about this topic, feel free to contact me.</p>
